cmd/clusterctl/examples/openstack/provider-component/user-data/centos/templates/master-user-data.sh

```diff
@@ -58,25 +57,31 @@ gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cl
 exclude=kube*
 EOF
 
-setenforce 0
-yum install -y kubelet-$CONTROL_PLANE_VERSION kubeadm-$CONTROL_PLANE_VERSION kubectl-$CONTROL_PLANE_VERSION --disableexcludes=kubernetes
+if [[ $(getenforce) != 'Disabled' ]]; then
+  setenforce 0
+fi
+
+yum install -y kubelet-$KUBELET_VERSION kubeadm-$KUBELET_VERSION kubectl-$KUBELET_VERSION --disableexcludes=kubernetes
 
 function install_configure_docker () {
     # prevent docker from auto-starting
     echo "exit 101" > /usr/sbin/policy-rc.d
     chmod +x /usr/sbin/policy-rc.d
     trap "rm /usr/sbin/policy-rc.d" RETURN
     yum install -y docker
-    echo 'OPTIONS="--selinux-enabled --log-driver=journald --signature-verification=false --iptables=false --ip-masq=false"' >> /etc/sysconfig/docker
+    echo 'OPTIONS="--log-driver=journald --signature-verification=false --iptables=false --ip-masq=false"' >> /etc/sysconfig/docker
     systemctl daemon-reload
     systemctl enable docker
     systemctl start docker
 }
 
 install_configure_docker
 
+# Get docker cgroup driver for kubelet configuration
+CG_DRIVER=$(docker info --format '{{.CgroupDriver}}')
+
 cat <<EOF > /etc/default/kubelet
-KUBELET_KUBEADM_EXTRA_ARGS=--cgroup-driver=systemd
+KUBELET_KUBEADM_EXTRA_ARGS=--cgroup-driver=$CG_DRIVER
```

Comment:
@hchenxa would you please create another PR by adding this logic to ubuntu and coreos as well? Enable kubelet use same cgroup driver as docker.

Reply:
ok.